products/BellHybrid/apps/application-bell-main/windows/BellHomeScreenWindow.cpp

```diff
@@ -91,182 +95,153 @@ namespace gui
+            removeWidget(currentLayout->getLayout());
+        }
+        currentLayout = std::move(layout);
+        addWidget(static_cast<BellBaseLayout *>(currentLayout->getLayout()));
+        presenter->onBeforeShow();
+
+        if (auto snoozeTimer = currentLayout->getSnoozeTimer()) {
+            auto timer = std::make_unique<app::ProgressTimerWithSnoozeTimer>(
+                application, *this, snoozeTimerName, timerTick, app::ProgressCountdownMode::Increasing);
+            timer->attach(snoozeTimer);
+            presenter->setSnoozeTimer(std::move(timer));
+        }
     }
 
     void BellHomeScreenWindow::setAlarmTriggered()
     {
-        alarm->setAlarmStatus(AlarmSetSpinner::Status::RINGING);
+        if (currentLayout) {
+            currentLayout->setAlarmTriggered();
+        }
     }
 
     void BellHomeScreenWindow::setAlarmActive(bool val)
     {
-        if (val) {
-            alarm->setAlarmStatus(AlarmSetSpinner::Status::ACTIVATED);
-        }
-        else {
-            alarm->setAlarmStatus(AlarmSetSpinner::Status::DEACTIVATED);
+        if (currentLayout) {
+            currentLayout->setAlarmActive(val);
         }
     }
 
-    void BellHomeScreenWindow::setHeaderViewMode(app::home_screen::HeaderViewMode mode)
+    void BellHomeScreenWindow::setViewState(app::home_screen::ViewState state)
     {
-        switch (mode) {
-        case app::home_screen::HeaderViewMode::Empty:
-            alarm->setVisible(false);
-            alarm->setAlarmTimeVisible(false);
-            snoozeTimer->setVisible(false);
-            alarm->informContentChanged();
-            break;
-        case app::home_screen::HeaderViewMode::AlarmIconAndTime:
-            alarm->setVisible(true);
-            alarm->setAlarmTimeVisible(true);
-            snoozeTimer->setVisible(false);
-            alarm->informContentChanged();
-            break;
-        case app::home_screen::HeaderViewMode::AlarmIcon:
-            alarm->setVisible(true);
-            alarm->setAlarmTimeVisible(false);
-            snoozeTimer->setVisible(false);
-            alarm->informContentChanged();
-            break;
-        case app::home_screen::HeaderViewMode::SnoozeIconAndTime:
-            alarm->setVisible(false);
-            alarm->setAlarmTimeVisible(false);
-            snoozeTimer->setVisible(true);
-            snoozeTimer->setTimerVisible(true);
-            snoozeTimer->informContentChanged();
-            break;
-        case app::home_screen::HeaderViewMode::SnoozeIcon:
-            alarm->setVisible(false);
-            alarm->setAlarmTimeVisible(false);
-            snoozeTimer->setVisible(true);
-            snoozeTimer->informContentChanged();
-            snoozeTimer->setTimerVisible(false);
-            break;
+        if (currentLayout) {
+            currentLayout->setViewState(state);
         }
     }
 
     void BellHomeScreenWindow::setTemperature(utils::temperature::Temperature newTemp)
     {
-#if CONFIG_ENABLE_TEMP == 1
```

Comment:
Will this flag still work?

Reply:
no, I will delete that

Reply:
Is a similair mechanism added in custom homescreens so we can still hide this feature?

Reply:
basically we can just disable the layout with temperature from selection screen - I think that will make more sense

Reply:
Then maybe we should add this disable in this step to avoid loosing this functionality as we don't want to show temeprature reading yet.

Reply:
that needs to be done in BH-1314 since this is only the "backend" mechanism to change homescreen and setting specific one in settings will be done in BH-1314